File: pro/lab_3/lab_3_c/inout.c
#include "struct.h"
#include "inout.h"
#include "time.h"

//input martix

void creat_sorted_array(matrix *matr){
	int i, j;
	double elem = -N*M;
	srand(time(NULL));
	for(i = 0; i < M; i++)
		for(j = 0; j < N; j++){
			elem += rand()%3;
			matr->mas[i][j] = elem / 10;
		}
}

void creat_inversely_sorted_array(matrix *matr){
	int i, j;
	double elem  = N*M;
	srand(time(NULL));
	for(i = 0; i < M; i++)
		for(j = 0; j < N; j++){
			elem -= rand()%5;
			matr->mas[i][j] = elem / 10;
		}
}

void creat_randomly_sorted_array(matrix *matr){
	int i, j;
	srand(time(NULL));
	for(i = 0; i < M; i++)
		for(j = 0; j < N; j++)
			matr->mas[i][j] = (rand()%(2*N*M) - N*M) / 10.0;
}

void creat_keyboard_input(matrix *matr){
	printf("Enter elements the matrix [%d][%d]:\n", M, N);
	int i, j;
	for(i = 0; i < M; i++)
		for(j = 0; j < N; j++)
			scanf("%lf", &(matr->mas[i][j]));
}

//print matrixt

void print_matrix(matrix *matr){
	int i, j;
	printf("Print matrix [%d][%d]:\n", M, N);
	for(i = 0; i < M; i++){
		for(j = 0; j < N; j++)
			printf("%5.1lf", matr->mas[i][j]);
		printf("\n");
	}
	printf("\n");
}

void print_first_positive_element(matrix *matr){
	search_first_positive_element(matr);
	
	if (matr->first_positive_element < 0)
		printf("Not found positive element!\n");
	else
		printf("First positive element [%d][%d]: %.1lf\n\n",
			matr->coord_i, matr->coord_j, matr->first_positive_element);
}

File: pro/lab_2/java/_inout/inout.java
package _inout;

import java.util.Scanner;
import _struct.*;
import _algo.*;

public class inout{

//input matrix
	public void creat_sorted_array(struct matr){
		int i, j;
		int elem = -matr.N*matr.M;
		for(i = 0; i < matr.M; i++)
			for(j = 0; j < matr.N; j++){
				elem += (i*j+i+j+i)%3;
				matr.mas[i][j] = elem / 10.0;
		}
	}

	public void creat_inversely_sorted_array(struct matr){
		int i, j;
		double elem  = matr.N*matr.M;
		for(i = 0; i < matr.M; i++)
			for(j = 0; j < matr.N; j++){
				elem -= (i*j+j+i)%5;
				matr.mas[i][j] = elem / 10;
			}
	}

	public void creat_randomly_sorted_array(struct matr){
		int i, j;
		for(i = 0; i < matr.M; i++)
			for(j = 0; j < matr.N; j++)
				matr.mas[i][j] = ((2*matr.N*matr.M*i)%15 - (matr.N*matr.M)%(i+j+5)) / 10.0;
	}
  
	public void creat_keyboard_input(struct matr){
		System.out.printf("Enter elements the matrix [%d][%d]:\n", matr.M, matr.N);
		int i, j;
		Scanner in = new Scanner(System.in);
		for(i = 0; i < matr.M; i++)
			for(j = 0; j < matr.N; j++)
				matr.mas[i][j] = in.nextDouble();
	}


// output matrix
	public void print_matrix(struct matr){
		int i, j;
		System.out.printf("Print matrix [%d][%d]:\n", matr.M, matr.N);
		for(i = 0; i < matr.M; i++){
			for(j = 0; j < matr.N; j++)
				System.out.printf("%5.1f", matr.mas[i][j]);
			System.out.printf("\n");
		}
		System.out.printf("\n");
	}

	public void print_first_positive_element(struct matr){
		algo al = new algo();
		al.search_first_positive_element(matr);
	
		if (matr.first_positive_element < 0)
			System.out.printf("Not found positive element!\n");
		else
			System.out.printf("First positive element [%d][%d]: %.1f\n\n",
				matr.coord_i, matr.coord_j, matr.first_positive_element);
	}


}